### PR TITLE
DOC: NDFrame.to_hdf(data_columns) documented (#13061).

### LINKED ISSUE #13061: DOC: DataFrame.to_hdf() docstring does not mention data_columns
#### Code Sample, a copy-pastable example if possible
To be able to do on-disk filtering per column, one needs to store not only in `table` format, but also create an index in the HDF file for the columns required to be indexed, or setting it to `True`, for all columns to be indexed.
This needs to be done by using the method parameter `data_columns`, which is completely missing from the docstring of `to_hdf`:
```python
df.to_hdf('teststore.hdf', 'df',format='t', data_columns=True)
```
Docstring is [here](https://github.com/pydata/pandas/blob/e462ece64b7cb12fe7cad6e085f62f3bfe9785c6/pandas/core/generic.py#L1053-L1093)

#### output of ``pd.show_versions()``

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.1.final.0
python-bits: 64
OS: Darwin
OS-release: 14.5.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.18.0
nose: None
pip: 8.1.1
setuptools: 20.7.0
Cython: 0.24
numpy: 1.11.0
scipy: 0.17.0
statsmodels: None
xarray: 0.7.1
IPython: 4.2.0
sphinx: 1.4.1
patsy: None
dateutil: 2.5.2
pytz: 2016.3
blosc: None
bottleneck: None
tables: 3.2.2
numexpr: 2.5.2
matplotlib: 1.5.1
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.8
boto: None

### PR DESCRIPTION
**Pull request by jzwinck**
- [x] closes #13061
- [ ] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

I'm aware there's another PR for the same issue, but it hasn't had activity in a while so this is my attempt.  I took the opportunity to remove the unusable "mode=r" option from the same docstring, and make the title clearer.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -1126,7 +1124,9 @@ def to_hdf(self, path_or_buf, key, **kwargs):
             If applying compression use the fletcher32 checksum
         dropna : boolean, default False.
             If true, ALL nan rows will not be written to store.
-
+        data_columns : list of columns to create as data columns, or True to
+            use all columns.  If True and format='table', data will be written
```
Comment (jreback) [inline pandas/core/generic.py]: this last line is superfluous. This and HDFStore.append for data_columns should be the same. pls add a link to the docs in both.

Comment (jzwinck) [inline pandas/core/generic.py]: @jreback Can you clarify what you mean by "add a link to the docs in both"?

Comment (jreback) [inline pandas/core/generic.py]: a ref to: http://pandas.pydata.org/pandas-docs/stable/io.html#query-via-data-columns

and ideally adding references as needed (e.g. compression) and for the format option

**Comment by jzwinck** [comment]
@jreback Can you review this again?  I've updated it as per your previous comments.

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**